From 1060c1d50e51b23dd1bacb56aabfb0adab4904f0 Mon Sep 17 00:00:00 2001
From: Mattias Busck <mattias.busck@hostmobility.com>
Date: Wed, 25 Jan 2023 11:31:52 +0100
Subject: [PATCH] HMX:DT Fixed T1 PHYs on 1Gbps

Tested with tools at
git@gitlab.com:hostmobility/hmx.git/testing/t1_phys

gpioset --mode=time --sec=1 $(gpiofind A_ETH_nRST)=1
gpioset --mode=time --sec=1 $(gpiofind A_ETH_nRST)=0
./test2_cable_t1_loop.py
---
 .../dts/freescale/imx8mp-var-dart-hmx1.dts    | 69 ++++++++++++++-----
 1 file changed, 53 insertions(+), 16 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts b/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts
index ed20a2c76d8f..25c0f0877851 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts
@@ -9,6 +9,7 @@
 
 / {
 	model = "Host Mobility HMX1 imx8mp";
+	comment = "pull rgmii-rxid driveback GIGABIT eth1 fixed back wednesday 28 dec maybe not disable. phy 3/7 no reset";
 
 	aliases {
 		rtc0 = &rtc_extern;
@@ -247,17 +248,25 @@ can5_osc: can5_osc {
 &fec {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_fec>;
-	phy-mode = "rgmii";
-	phy-handle = <&ethphy_t1_1>;
+	phy-mode = "rgmii-rxid";
+	//phy-handle = <&ethphy_t1_1>;
 	status = "okay";
+	//phy-reset-gpios = <&gpio3 1 GPIO_ACTIVE_HIGH>;
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
 	mdio {
 		#address-cells = <1>;
 		#size-cells = <0>;
+		/* try with fixed link
 		ethphy_t1_1: ethernet-phy@0 {
 			// trying med clause 22 instead of clause 45
+			// qqq: check what happens now compatible = "ethernet-phy-ieee802.3-c45";
+			//compatible = "marvell";
 			compatible = "ethernet-phy-ieee802.3-c45";
 			reg = <3>;
-			reset-gpios = <&gpio3 1 GPIO_ACTIVE_HIGH>;
+			//reset-gpios = <&gpio3 1 GPIO_ACTIVE_HIGH>;
 			at803x,eee-disabled;
 			eee-broken-1000t;
 			reset-assert-us = <10000>;
@@ -269,18 +278,27 @@ vddio1: vddio-regulator {
 				regulator-max-microvolt = <1800000>;
 			};
 		};
+		*/
+		ethphy_t1_2: ethernet-phy@1 {
+			// trying med clause 22 instead of clause 45
+			compatible = "ethernet-phy-ieee802.3-c45";
+			reg = <7>;
+		};
 	};
 };
 
 /* Second Automotive Ethernet Marvell 88Q2110*/
 &eqos {
-	mdio {
-		ethernet-phy@0 {
-			// trying med clause 22 instead of clause 45
-			compatible = "ethernet-phy-ieee802.3-c45";
-			reg = <3>;
-		};
+	phy-mode = "rgmii-rxid";
+	phy-handle = <&ethphy_t1_2>;
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+	/* MAYBE NOT DISABLE mdio {
+		status = "disabled";
 	};
+	*/
 };
 
 &i2c2 {
@@ -631,6 +649,25 @@ &uart4 {
        status = "disabled";
 };
 
+/* From i.MX 8M Plus Applications Processor Reference Manual
+   8.2.4.205 SW_PAD_CTL_PAD_NAND_CE0_B SW PAD Control
+   Register (IOMUXC_SW_PAD_CTL_PAD_NAND_CE0_B) */
+
+#define PE_0_PULL_DISABLE (1<<8)
+#define PE_1_PULL_ENABLE (1<<8)
+#define HYS_0_CMOS (0<<7)
+#define HYS_1_SCHMITT (1<<7)
+#define PUE_0_WEAK_PULL_DOWN (0<<6)
+#define PUE_1_WEAK_PULL_UP (1<<6)
+#define ODE_0_OPEN_DRAIN_DISABLE (0<<5)
+#define ODE_1_OPEN_DRAIN_ENABLE (1<<5)
+#define FSEL_0_SLOW_SLEW_RATE (0<<4)
+#define FSEL_1_FAST_SLEW_RATE (1<<4)
+#define DSE_X1 (0<<1)
+#define DSE_X2 (2<<1)
+#define DSE_X4 (1<<1)
+#define DSE_X6 (3<<1)
+
 &iomuxc {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_hog>;
@@ -662,13 +699,13 @@ MX8MP_IOMUXC_SAI1_RXD6__ENET1_RGMII_RD2				0x90
 			MX8MP_IOMUXC_SAI1_RXD7__ENET1_RGMII_RD3				0x90
 			MX8MP_IOMUXC_SAI1_TXC__ENET1_RGMII_RXC				0x90
 			MX8MP_IOMUXC_SAI1_TXFS__ENET1_RGMII_RX_CTL			0x90
-			MX8MP_IOMUXC_SAI1_TXD0__ENET1_RGMII_TD0				0x00
-			MX8MP_IOMUXC_SAI1_TXD1__ENET1_RGMII_TD1				0x00
-			MX8MP_IOMUXC_SAI1_TXD2__ENET1_RGMII_TD2				0x00
-			MX8MP_IOMUXC_SAI1_TXD3__ENET1_RGMII_TD3				0x00
-			MX8MP_IOMUXC_SAI1_TXD4__ENET1_RGMII_TX_CTL			0x00
-			MX8MP_IOMUXC_SAI1_TXD5__ENET1_RGMII_TXC				0x00
-			MX8MP_IOMUXC_NAND_CE0_B__GPIO3_IO01				0x02
+			MX8MP_IOMUXC_SAI1_TXD0__ENET1_RGMII_TD0				0x16
+			MX8MP_IOMUXC_SAI1_TXD1__ENET1_RGMII_TD1				0x16
+			MX8MP_IOMUXC_SAI1_TXD2__ENET1_RGMII_TD2				0x16
+			MX8MP_IOMUXC_SAI1_TXD3__ENET1_RGMII_TD3				0x16
+			MX8MP_IOMUXC_SAI1_TXD4__ENET1_RGMII_TX_CTL			0x16
+			MX8MP_IOMUXC_SAI1_TXD5__ENET1_RGMII_TXC				0x16
+			MX8MP_IOMUXC_NAND_CE0_B__GPIO3_IO01		(PE_1_PULL_ENABLE  | PUE_0_WEAK_PULL_DOWN | DSE_X2)
 
 		>;
 	};
-- 
2.30.2

